src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -241,6 +244,45 @@ public IActionResult ApplyConfigSettings(string sessionId, [FromBody] ApplyConfi
             return Ok(output);
         }
 
+        [HttpGet("session/<sessionId>/settings/<configSettingId>/resources")]
+        [SwaggerOperation(OperationId = "GetConfigSettingResources")]
+        [SwaggerResponse(200, type: typeof(GetConfigSettingResourcesOutput))]
+        [Authorize]
+        public async Task<IActionResult> GetConfigSettingResources(string sessionId, string configSettingId)
+        {
+            var state = _stateServer.Get(sessionId);
+            if (state == null)
+            {
+                return NotFound($"Session ID {sessionId} not found.");
+            }
+
+            var serviceProvider = CreateSessionServiceProvider(state);
+            var typeHintCommandFactory = serviceProvider.GetRequiredService<ITypeHintCommandFactory>();
+
+            if (state.SelectedRecommendation == null)
```

Comment:
nit: Do this right after the `if` block on 254. Seems like we should start the method validating state. Once that is validated then create the session service provider and such.

Reply:
Done